File: src/tests/DataBlock Sample Application/VB.NET/Territories/TerritoriesTableMetadata.vb



Imports System
Imports System.Data
Imports voidsoft.DataBlock






Namespace ExtenderTerritories

    Public Class TerritoriesTableMetadata
        Inherits TableMetadata
        Private _fields As DatabaseField()

        Public Sub New()
            _fields = New DatabaseField(2) {}
            _fields(0) = New DatabaseField(DbType.String, "TerritoryID", True, False, Nothing)
            _fields(1) = New DatabaseField(DbType.String, "TerritoryDescription", False, False, Nothing)
            _fields(2) = New DatabaseField(DbType.Int32, "RegionID", False, False, Nothing)
            Me.TableName = "Territories"
        End Sub

        Public Overloads Overrides Property TableFields() As DatabaseField()
            Get
                Return _fields
            End Get
            Set(ByVal Value As DatabaseField())
                _fields = Value
            End Set
        End Property

        Public Property TerritoryID() As System.String
            Get
                Return CType((Me.GetField("TerritoryID")).fieldValue, System.String)
            End Get
            Set(ByVal Value As System.String)
                Me.SetFieldValue("TerritoryID", Value)
            End Set
        End Property

        Public Property TerritoryDescription() As System.String
            Get
                Return CType((Me.GetField("TerritoryDescription")).fieldValue, System.String)
            End Get
            Set(ByVal Value As System.String)
                Me.SetFieldValue("TerritoryDescription", Value)
            End Set
        End Property

        Public Property RegionID() As System.Int32
            Get
                Return CType((Me.GetField("RegionID")).fieldValue, System.Int32)
            End Get
            Set(ByVal Value As System.Int32)
                Me.SetFieldValue("RegionID", Value)
            End Set
        End Property
    End Class
End Namespace

File: src/tests/Console.Tests/TerritoriesTableMetadata.cs
using System;
using System.Data;
using voidsoft.DataBlock;

namespace ExtenderTerritories
{


		public class TerritoriesTableMetadata : TableMetadata
		{
			private DatabaseField[] _fields;

			public TerritoriesTableMetadata()
			{
					_fields = new DatabaseField[3];
 _fields[0] = new DatabaseField(DbType.String,"TerritoryID",true, false,null);
 _fields[1] = new DatabaseField(DbType.String,"TerritoryDescription",false, false,null);
 _fields[2] = new DatabaseField(DbType.Int32,"RegionID",false, false,null);
 
this.currentTableName = "Territories";
}


			public override DatabaseField[] TableFields 
			{
				get{ return _fields;}
				set{_fields = value;}
			}

public System.String TerritoryID
{
    get
    {
          return (System.String) (this.GetField("TerritoryID")).fieldValue;
    }

    set
    {
          this.SetFieldValue("TerritoryID", value);
    }
}


public System.String TerritoryDescription
{
    get
    {
          return (System.String) (this.GetField("TerritoryDescription")).fieldValue;
    }

    set
    {
          this.SetFieldValue("TerritoryDescription", value);
    }
}


public System.Int32 RegionID
{
    get
    {
          return (System.Int32) (this.GetField("RegionID")).fieldValue;
    }

    set
    {
          this.SetFieldValue("RegionID", value);
    }
}

}
}
